### PR TITLE
User-referral for originprotocol.com

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

This is the marketing component and mobile gap-bridge for user-referrals.

**NOTE**: This depends on some changes to the growth server.  Ideally that would be merged and deployed before this.

### Checklist:

- [ ] Test your work and double-check to confirm that you didn't break anything
- [ ] Update the README, if necessary
- [ ] Run [translation commands](https://github.com/OriginProtocol/origin-website/tree/master/translations#extract-newedited-english-strings-for-translation) if any strings are changed

### CODE REVIEW

**Comment by mikeshultz** [comment]
Blocked by https://github.com/OriginProtocol/origin/pull/4071

File: templates/referral_campaign.html [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+      colorLight : "#ffffff",
+      correctLevel : QRCode.CorrectLevel.H
+    }
+    const qrcode1 = new QRCode('qrcode1', qrOptions)
+    const qrcode2 = new QRCode('qrcode2', qrOptions)
+
+    /**
+     * Copy a string to the clipboard
+     */
+    function codeToClipboard(code) {
```
Comment (sparrowDom) [inline templates/referral_campaign.html]: nit: a lot of this code is already in the [partner_campaign](https://github.com/OriginProtocol/origin-website/blob/48ff064f9dfbca2920690c32ead51027a6918974/templates/partner_campaign.html) Could be separated into a third file and imported from there into this and `partner_campaign` file.

File: templates/referral_campaign.html [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+        if (!win) {
+          window.location = url
+        }
+      }
+    }
+
+    /**
+     * Figure out which button to display
+     */
+    function detectButtion(code) {
```
Comment (sparrowDom) [inline templates/referral_campaign.html]: nit: typo

File: templates/referral_campaign.html [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+      if (!code) {
+        console.error('no code provided to detectButtion')
```
Comment (sparrowDom) [inline templates/referral_campaign.html]: typo

File: templates/referral_campaign.html (hunk 1 above)
Comment (mikeshultz) [reply templates/referral_campaign.html]: Went for speed, not cleanliness.  If these look like they'll stick around, I'll take the time to clean it up a bit.

File: templates/referral_campaign.html (hunk 2 above)
Comment (mikeshultz) [reply templates/referral_campaign.html]: Butt ions are important charged atoms!

File: templates/referral_campaign.html (hunk 1 above)
Comment (sparrowDom) [reply templates/referral_campaign.html]: sounds good 👍

**Pull request merged**